src/haddock/modules/analysis/ilrmsdmatrix/__init__.py

```diff
@@ -205,29 +214,58 @@ def _run(self):
             nmodels,
             tot_npairs,
             ncores)
-
-        # Calculate the ilrmsd for each set of models
-        ilrmsd_jobs = []
-        self.log(f"running ilRmsd Jobs with {ncores} cores")
+        
+        rec_traj_filename = "traj_rec.xyz"
+        lig_traj_filename = "traj_lig.xyz"
+        prev_keys = []
+        with open(rec_traj_filename, "w") as rec_traj_xyz:
+            with open(lig_traj_filename, "w") as lig_traj_xyz:
+                for mod in models:
+                    atoms = {}
+                    atoms.update(get_atoms(mod))
+                    ref_coord_dic, _ = load_coords(
+                    mod, atoms, res_resdic
+                    )
+                    if prev_keys != []:
+                        if ref_coord_dic.keys() != prev_keys:
+                            self.finish_with_error(
+                                "The keys of the ref_coord_dic are not the "
+                                "same for all the models. Please check the "
+                                "input models."
+                                )
+                    # write receptor coords
+                    rec_coords = {k:ref_coord_dic[k] for k in ref_coord_dic if k[0] == self.params["receptor_chain"]}
+                    lig_coords = {k:ref_coord_dic[k] for k in ref_coord_dic if k[0] in self.params["ligand_chains"]}
+                    n_rec_atoms = len(rec_coords)
+                    n_lig_atoms = len(lig_coords)
+                    rec_traj_xyz.write(f"{n_rec_atoms}{os.linesep}{os.linesep}")
+                    lig_traj_xyz.write(f"{n_lig_atoms}{os.linesep}{os.linesep}")
+                    for k, v in rec_coords.items():
+                        rec_traj_xyz.write(f"x {v[0]} {v[1]} {v[2]}{os.linesep}")
+                    for k, v in lig_coords.items():
+                        lig_traj_xyz.write(f"x {v[0]} {v[1]} {v[2]}{os.linesep}")
```

Comment:
`k` is not used here, best to loop over the values intstead

```suggestion
                    for v in rec_coords.values():
                        rec_traj_xyz.write(f"x {v[0]} {v[1]} {v[2]}{os.linesep}")
                    for v in lig_coords.values():
                        lig_traj_xyz.write(f"x {v[0]} {v[1]} {v[2]}{os.linesep}")
```